mylist.cpp

```diff
@@ -193,38 +232,76 @@ void mylist::removefront()
 // if the list becomes empty, head, tail, and size must be set to
 // the state of an empty list.
 //
-void mylist::removeback()
+void mylist::removeback() // my Part
 {
-  // change me, your code here
-  cout << "removeback is not supported.\n";
+  if(tail == NULL){
+    throw"removefront: Can't remove from an empty list.";
+  }
+  if(head == tail){
+    delete head;
+    head = tail = NULL;
+    size = 0;
+  } else {
+    listnode *t = tail;
+    tail = tail->prev;
+    tail->next = NULL;
+    delete t;
+    size--;
+  }
 }
 
 //
 // remove the node pointed to by ptr
 // if the list becomes empty, head, tail, and size must be set to
 // the state of an empty list.
 //
-void mylist::remove(listnode *ptr)
+void mylist::remove(listnode *ptr) // my part
 {
   if (ptr == NULL) {
-    cout << "Remove: can't remove an NULL pointer.\n";
-    exit(0);
+    throw  "Remove: can't remove an NULL pointer.";
+  }
+  if(ptr == head){
+    removefront();
+  } else if(ptr == tail){
+    removeback();
+  } else{
+    listnode *t = head;
+    while(t != NULL){
```

Comment:
I think this function should remove only one node,  but it deletes more than one.